kumascript/macros/AvailableInWorkers.ejs

```diff
@@ -3,9 +3,15 @@
 //
 // Parameters:
 //
-//  $0 - workerType (optional)
+//  $0 - workerType (optional):
+//      'only_dedicated_and_window': only in DedicatedWorker (and in Window)
+//      'only_dedicated' only in DedicatedWorker
+//      'except_service': all workers but ServiceWorker (and in Window)
+//      'except_service_and_window': all workers but ServiceWorker (and no window)
```

Comment:
@teoli2003 Isn't this the same as `only_dedicated`?

Reply:
No, there can be a `SharedWorker` too: https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker

(At least it is my understanding)

Reply:
except_service_and_window = DedicatedWorker + SharedWorker
only_dedicated = DedicatedWorker

Reply:
Thanks for the clarification, so this sounds like `except_service_and_window` could also be named `dedicated_and_shared`?

Wouldn't it be easiest to just list the workers in which the feature is available (e.g. `dedicated,shared`) and display a comma separated list of worker types where the feature is available?

Looking at the screenshots again, I find it confusing that there is "available in Dedicated Web Workers" and "only available in Dedicated Web Workers".

Reply:
@caugner it sounds like this is hard because we can't easily specify multiple options in a macro. Why don't we just do the same as compat/specs and pull the information from metadata? Then we can make the support explicit:

worker support
- window
- dedicated worker
- shared worker
- service worker

Then `{{AvailableInWorkers}}` renders some standard text then "Supported in the following types of workers: [list]". If no metadata specified we could assume "all".

Would that be better?

Reply:
We can get the info from w3c/webref. Nevertheless, there is one caveat the website equivalent for `{{AvailableInWorker}}`: it is more generic as would really be an `{{Availability}}` macro. There are many more cases: it would tell us if something is available in Worklet (of all kinds), in RTC's Identity provider, …

I think the mid-term solution would be to ditch `{{AvailableInWorker}}` for an `{{Availability}}` macro that we can put everywhere (like `{{Specifications}}` and `{{Compat}}`). I didn't do this here as this is a much more significant chunk of work (because there are many more cases, and it has to work with additional ones that can appear in the future, and because we need to obtain the w3c/webref key – again not really difficult but a more significant chunk of work).

That's why I wanted a short-term, more effortless solution (once we have the mid-term solution, a script can be run to update all the pages).

From the two proposals from @caugner:
- more explicit text (`dedicated_and_shared`)
- list of features

I prefer the first one as we don't need to add some parsing (or a loop with many unused cases) to the code. The only caveat is that it won't be compatible with the current arguments (which is why I chose the current keywords), so we need a find and replace PR on the whole MDN once this lands (not difficult).

It is on my todo list to update this PR, hopefully today (but so many things are happening in parallel, so that's not a promise)

Reply:
FWIW I much prefer my proposal of using metadata.  This https://github.com/mdn/yari/pull/10029#issuecomment-1849511648 is overly complicated because you're trying to define exceptions.

I'm not sure about "one `{{Availability}}` macro to rule them all". If we have stuff in metadata I think we can do away with macros altogether at some point.

Reply:
In the mid-term, we won't even need metadata as this information is available via `w3c/webref`, which we already use in some macros. (But that's a significant chunk of work, and we need both some writers to define where and how (=text) to display on the page and non-trivial reviewing of the new macro).

That's why I would prefer this stop-gap first: it is similar to what we do know and doesn't impair the change in the future (a find-and-replace script will be easy to do)

Reply:
OK. Then how do we get this in!!!